lib/Sema/SemaDecl.cpp

```diff
@@ -10868,6 +10868,30 @@ void Sema::ActOnFinishKNRParamDeclarations(Scope *S, Declarator &D,
   }
 }
 
+/// ActOnBoundsExpr: attach a bounds expression to a parameter declaration.
+void Sema::ActOnBoundsExpr(Decl *D, Expr *BoundsExpr) {
+  if (!D || !BoundsExpr)
+    return;
+
+  VarDecl *Var = cast<VarDecl>(D);
```

Comment:
This cast asserts in debug builds, but that leaves release builds at risk.  We could use dyn_cast instead for release builds.